### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by anas599**
In this PR:

- Create a class Classroom with @label instance variable
- Create the has-many/belongs-to relationship between Classroom and Student
- Create a class Book with @title and @author  instance variable
- Create a class Rental with @date  instance variable
- Create the many-to-many relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: lib/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  attr_accessor :name, :students, :label
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    students.push(student)
+  end
```
Comment (Mihndim2020) [inline lib/classroom.rb]: - [ ] When adding a student to a classroom it also sets the classroom for the student, something like so: 

```ruby
  def add_student(student)
    @students << student
    student.classroom = self
  end
```
Please kindly fix that

File: lib/student.rb [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+
 class Student < Person
-  def initialize(classroom, age, name = 'Unknown')
+  attr_accessor :name, :classroom
+
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
     super(name, age, parent_permission)
     @classroom = classroom
+    classroom.students << self
   end
```
Comment (Mihndim2020) [inline lib/student.rb:11]: - [ ] There should be a method that serves as the 'belongs-to' side (`a student belongs to a classroom`), something like so: 

```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @anas599, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @anas599 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**